streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -107,11 +107,11 @@ public void onCompletion(final RecordMetadata metadata, final Exception exceptio
                         } else {
                             if (sendException == null) {
                                 sendException = exception;
-                                if (sendException instanceof ProducerFencedException) {
-                                    log.error("{} Error sending record to topic {}. No more offsets will be recorded for this task and it will be closed as it is a zombie.", logPrefix, topic, exception);
-                                } else {
-                                    log.error("{} Error sending record to topic {}. No more offsets will be recorded for this task and the exception will eventually be thrown", logPrefix, topic, exception);
-                                }
+                                log.error("{} Error sending record with key {} to topic {}, partition {}. " +
```

Comment:
I can change back, but it's not great practice to have a lower layer assume what a higher layer will do with the exception. For example, if later we decide to do something different with fenced exceptions, we'll then have to update this comment as well. I'd rather log a generic message here (without the "eventually be thrown" part)